google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +135,33 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    while (throwable != null) {
+      if (throwable.getMessage().contains("Connection closed prematurely")
+          || throwable.getCause() instanceof javax.net.ssl.SSLException
```

Comment:
Is there any particular reason to use fully qualified names here? If no, please change to importing. Also, please consider assigning `Throwable cause = throwable.getCause()` and than use it like `cause instanceof SSLException`, just to make the huge if statement less scary (if you choose to do so, please also change line 148 to `throwable = cause`). 

Also, even though it is very unlikely, but there is a small chance that throwable-cause may form an infinite loop (if any child eventually becomes a parent of any of its ancestors).